### PR TITLE
RM-113 remove upper bound pyyaml

### PR DESCRIPTION
**Pull request by ryantimjohn**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -298,15 +298,7 @@ def initialize_options(self) -> None:
...
-          # awscli pins PyYAML. We have seen issues at
-          # run-time if we don't constrain things here as well. So
-          # this tracks the awscli req for python >= 3.6 which is
-          # what we support
-          #
-          # https://github.com/aws/aws-cli/blob/develop/setup.py
-          # #### NEW UNFORTUNATE CONSTRAINTS #####
-          # https://github.com/yaml/pyyaml/issues/724
-          'PyYAML>=3.10,<=5.3.1',
+          'PyYAML>=3.10',
```
Comment (Brunope) [inline setup.py:301]: pyyaml was one of the packages i remember that was causing problems on some specific versions when I was recently fixing homebrew stuff. which version did this test run with?

Comment (ryantimjohn) [reply setup.py:301]: This ran across 3.8, 3.9, 3.10, and 3.11

Comment (ryantimjohn) [reply setup.py:301]: Do you have any logs from the home brew failure? Also, sorry, I didn't purposefully merge this but seems like I merged the PR after it so this merged 🙃

Comment (Brunope) [reply setup.py:301]: no which version of pyyaml i mean. but anyway what i ended up having to do was increase the pyyaml version to fix the brew build so i think this is fine

**Pull request merged**